internal/pkg/cli/merger/merger.go

```diff
@@ -60,13 +62,29 @@ func (p *Merger) Run() error {
 		}
 	}
 
+	baseProfile := contents[0].DeepCopyObject()
+
 	merged, err := recordingmerger.MergeProfiles(contents)
 	if err != nil {
 		return fmt.Errorf("merge profiles: %w", err)
 	}
 
-	var buffer bytes.Buffer
 	printer := printers.YAMLPrinter{}
+
+	if p.options.check {
+		if reflect.DeepEqual(baseProfile, merged) {
+			log.Println("Base profile is up-to-date.")
+			os.Exit(0)
+		} else {
+			log.Println("Base profile needs an update.")
+			if err := printer.PrintObj(merged, os.Stderr); err != nil {
+				return fmt.Errorf("print YAML: %w", err)
+			}
+			os.Exit(1)
```

Comment:
Not super happy with this here with straight up calling `os.Exit`, but returning an errors prints a slightly confusing message. Do we think this is appropriate here, or do you folks have better ideas? :)

Reply:
`log.Fatal` also does an `os.Exit(1)`, so I assume that's fine.